pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -2014,19 +2020,21 @@ func TestUpdateTaskRunResult2(t *testing.T) {
 func TestUpdateTaskRunResultTwoResults(t *testing.T) {
 	for _, c := range []struct {
 		desc          string
-		podStatus     corev1.PodStatus
+		pod           corev1.Pod
 		taskRunStatus *v1beta1.TaskRunStatus
 		want          []v1beta1.TaskRunResult
 	}{{
 		desc: "two test results",
-		podStatus: corev1.PodStatus{
-			ContainerStatuses: []corev1.ContainerStatus{{
-				State: corev1.ContainerState{
-					Terminated: &corev1.ContainerStateTerminated{
-						Message: `[{"key":"resultNameOne","value":"resultValueOne", "type": "TaskRunResult"},{"key":"resultNameTwo","value":"resultValueTwo", "type": "TaskRunResult"}]`,
+		pod: corev1.Pod{
+			Status: corev1.PodStatus{
+				ContainerStatuses: []corev1.ContainerStatus{{
+					State: corev1.ContainerState{
+						Terminated: &corev1.ContainerStateTerminated{
+							Message: `[{"key":"resultNameOne","value":"resultValueOne", "type": "TaskRunResult"},{"key":"resultNameTwo","value":"resultValueTwo", "type": "TaskRunResult"}]`,
```

Comment:
Can we add a test with duplicate keys here like you have in the issue to make sure that they are acutally deduplicated before updating the status `taskRun.Status.TaskRunResults`? feel free to ignore it if its not applicable ...

Reply:
@pritidesai Make sense, added ,thanks.